extensions/standard-processors/processors/GenerateFlowFile.cpp

```diff
@@ -77,88 +73,67 @@ void GenerateFlowFile::initialize() {
   setSupportedRelationships(relationships);
 }
 
-void GenerateFlowFile::onTrigger(core::ProcessContext *context, core::ProcessSession *session) {
-  uint64_t batchSize = 1;
-  bool uniqueFlowFile = true;
-  uint64_t fileSize = 1024;
-  bool textData = false;
+void generateData(std::vector<char>& data, bool textData = false) {  // NOLINT
+  std::random_device rd;
+  std::mt19937 eng(rd());
+  if (textData) {
+    std::uniform_int_distribution<> distr(0, TEXT_LEN);
```

Comment:
I know this is legacy, but it took me a while to figure out why this is right. The `TEXT_CHARS` array contains 91 elements, so `TEXT_LEN` in reality is not the length of the array, but the maximum index. I think the previous version, which used % TEXT_LEN was erroneus, and now that we generate indexes from [0, TEXT_LEN] we are good, but it is not easy to understand.
I'm OK with merging this, but it should be replaced with something more straightforward eventually.

Reply:
integer module should be 0 to n-1 inclusive, so erroneous because it's a max index and not a length? I don't necessarily care what the hold code did but this block does need some comments to clarify the change why it is how it is and if the names are misleading a clarification to the reason why it was wrong previously.

Reply:
I replaced the array with a C string, so we don't need the length any more.